MapboxSceneKit/Routes/Shaders/unlit-line.metal

```diff
@@ -0,0 +1,141 @@
+//
+//  unlit-line.metal
+//  line-rendering
+//
+//  Created by Jim Martin on 8/2/18.
+//  Copyright © 2018 Jim Martin. All rights reserved.
+//
+
+#include <metal_stdlib>
+
+using namespace metal;
+
+#include <SceneKit/scn_metal>
+
+struct NodeBuffer {
+    float4x4 modelTransform;
+    float4x4 modelViewTransform;
+    float4x4 normalTransform;
+    float4x4 modelViewProjectionTransform;
+};
+
+typedef struct {
+    float3 position [[ attribute(SCNVertexSemanticPosition) ]];
+    float3 normals [[ attribute(SCNVertexSemanticNormal) ]];
+    half4 color [[ attribute(SCNVertexSemanticColor)]];
+    float2 texCoords [[ attribute(SCNVertexSemanticTexcoord0) ]];
+    float2 lineParams [[ attribute(SCNVertexSemanticTexcoord1) ]];
+} VertexInput;
+
+struct Vertex
+{
+    float4 position [[position]];
+    half4 color;
+    float2 texCoords;
+    float capFlag;
+    float zOffset;
+};
+
+vertex Vertex lineVert(VertexInput in [[ stage_in ]],
+                       constant SCNSceneBuffer& scn_frame [[buffer(0)]],
+                       constant NodeBuffer& scn_node [[buffer(1)]])
+{
+    Vertex vert;
+    vert.position = scn_node.modelViewProjectionTransform * float4(in.position, 1.0);
+    vert.texCoords = in.texCoords;
+    vert.color = in.color;
+    
+    //give useful names to line params
+    float lineRadius = in.lineParams.y;
+    bool shouldModifyDepthBuffer = in.lineParams.x > 0;
+    
+    //calculate the offset amounts
+    float expandDistance = lineRadius;
+    
+    //apply the offset
+    float4 neighborPos = scn_node.modelViewProjectionTransform * float4(in.normals, 1.0);
+    float3 pVec = (neighborPos.xyz / neighborPos.w) - (vert.position.xyz / vert.position.w);
+    vert.capFlag = step(.001, length(pVec));
+    
+    float2 perpVec = normalize(pVec).yx;
+    perpVec.y *= -1;
+    perpVec.xy *=  (2 * (in.texCoords.x - .5)) * (2 * (in.texCoords.y - .5));
+    
+    //1.95 because the caps look better slightly small
```

Comment:
1.95? Or 1.98? Either way, probably best to remove the number from the comment, to avoid inaccuracies.